packages/react-native/React/Fabric/Mounting/ComponentViews/Image/RCTImageComponentView.mm

```diff
@@ -86,6 +121,23 @@ - (void)updateState:(const State::Shared &)state oldState:(const State::Shared &
 
   if (!havePreviousData ||
       (newImageState && newImageState->getData().getImageSource() != oldImageState->getData().getImageSource())) {
+    const auto &imageProps = static_cast<const ImageProps &>(*_props);
+    if (!imageProps.defaultSources.empty()) {
+      UIImage *defaultImage = [self _getUIImageWithImageSource:imageProps.defaultSources.front()];
+
+      if (defaultImage) {
+        __weak RCTImageComponentView *weakSelf = self;
+        [self _setImage:defaultImage
+            setImageBlock:^(UIImage *finalImage) {
+              RCTImageComponentView *strongSelf = weakSelf;
+              if (strongSelf) {
+                if (!strongSelf->_imageView.image) {
```

Comment:
```suggestion
              if (strongSelf && !strongSelf->_imageView.image) {
```

but what if we have an `image` already? I mean, what happens if we change the imageSource prop?

Reply:
If we have an image already, we would not reset default image and just only to load new image source.